rust/worker/src/execution/orchestration/hnsw.rs

```diff
@@ -620,22 +621,22 @@ impl Component for HnswQueryOrchestrator {
                 }
             };
 
-        match IndexConfig::from_segment(&hnsw_segment, collection.dimension.unwrap()) {
-            Ok(index_config) => {
-                self.index_config = Some(index_config);
-
-                // Normalize the query vectors if we are using the cosine similarity
-                if self.index_config.as_ref().unwrap().distance_function == DistanceFunction::Cosine
-                {
-                    for query_vector in self.query_vectors.iter_mut() {
-                        *query_vector = normalize(query_vector);
-                    }
-                }
-            }
+        let distance_function = match distance_function_from_segment(&hnsw_segment) {
+            Ok(distance_function) => distance_function,
             Err(e) => {
                 terminate_with_error(self.result_channel.take(), e, ctx);
                 return;
             }
+        };
+        self.index_config = Some(IndexConfig::new(
+            collection.dimension.unwrap(),
```

Comment:
should we clean up this unwrap?

Reply:
Hmm, ideally should also change the protobuf response message type so that it is not optional. Seems like deferrable to another PR